libraries/rush-lib/src/cli/RushCommandLineAPI.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  CommandLineAction,
+  CommandLineParameter,
+  CommandLineParameterKind
+} from '@rushstack/ts-command-line';
+import { RushCommandLineParser } from './RushCommandLineParser';
+
+interface ICommandLineParameter {
+  /**
+   * The corresponding string representation of CliParameterKind
+   */
+  readonly kind: string;
+
+  /**
+   * The long name of the flag including double dashes, e.g. "--do-something"
+   */
+  readonly longName: string;
+
+  /**
+   * An optional short name for the flag including the dash, e.g. "-d"
+   */
+  readonly shortName?: string;
+
+  /**
+   * Documentation for the parameter that will be shown when invoking the tool with "--help"
+   */
+  readonly description: string;
+
+  /**
+   * If true, then an error occurs if the parameter was not included on the command-line.
+   */
+  readonly required?: boolean;
+}
+
+interface IRushCliJsonSpec {
+  actionName: string;
+  parameters: ICommandLineParameter[];
+}
+
+/**
+ * Information about the available CLI commands
+ *
+ * @beta
+ */
+export class RushCommandLineAPI {
+  public static getSpec(workspaceFolder: string): IRushCliJsonSpec[] {
```

Comment:
The `Cli` acronymn in `IRushCliJsonSpec` should maybe be expanded to `CommandLine`.  But this spec is not Rush's command line really -- it is the general API we were hoping to promote into `ts-command-line`. So maybe it can be just `ICommandLineSpecJson` or just `ICommandLineSpec` .  From our design discussion, these specs are not always specified as json, and the JSON serializable concept could be arguably inherent in being a "spec".

Reply:
Changed to ICommandLineSpec